kedro_partitioned/pipeline/multinode.py

```diff
@@ -485,20 +497,47 @@ def _filter_cached(
         else:
             return intersection
 
+    def is_list_of_lists(self, variable):
+        return isinstance(variable, list) and all(
+            isinstance(item, list) for item in variable
+        )
+
     @property
     def func(self) -> Callable:
         def fn(*args: Any) -> List[List[str]]:
             partitioneds, configurators = self._extract_args(args)
+            if self.split_function is None:
+                intersection = self._intersect_partitioneds(partitioneds)
+            elif callable(self.split_function):
+                intersection = self.split_function(partitioneds)
+            else:
+                raise TypeError(f"{self.split_function}` must be a callable.")
+
+            if self.is_list_of_lists(intersection):
+                output = []
+                for inter in intersection:
+                    inter = self._apply_filter(inter)
+                    inter = self._filter_cached(configurators, inter)
+                    inter = sorted(inter)
+                    output.append(
+                        [
+                            self._slice_partitions(inter, i)
+                            for i in range(self._slice_count)
+                        ]
+                    )
+                transposed = list(zip(*output))
 
-            intersection = self._intersect_partitioneds(partitioneds)
-            intersection = self._apply_filter(intersection)
-            intersection = self._filter_cached(configurators, intersection)
-            intersection = sorted(intersection)
+                # Aplatir chaque groupe transposé
+                return [sum(group, []) for group in transposed]
```

Comment:
Tbh, I don't think I got what this feature is doing. Can you explain what it does and give an example of usage?

Add some documentation for this function on the docstring too